api/cluster.go

```diff
@@ -96,7 +99,12 @@ func (s *Server) getData(ctx *middleware.Context, request models.GetData) {
 func (s *Server) indexDelete(ctx *middleware.Context, req models.IndexDelete) {
 	defs, err := s.MetricIndex.Delete(req.OrgId, req.Query)
 	if err != nil {
-		response.Write(ctx, response.NewError(http.StatusBadRequest, err))
+		// errors can be due to bad user input or corrupt index.
+		if strings.Contains(err.Error(), "Index is corrupt") {
+			response.Write(ctx, response.WrapError(err))
+		} else {
+			response.Write(ctx, response.NewError(http.StatusBadRequest, err.Error()))
+		}
```

Comment:
Returning an InternalServer error to the user tells them that our code is buggy and broken.  It should only be returned when a bug is encountered.

there are a lot more error types then just internalServer and badRequests and we already use them throughout the code base.  eg permisionDenined errors due to exceeding defined limits, timeouts etc..

I dont agree that we should mask the error messages returned from metrictank, especially as the codebase is constantly changing and immature.  Once the code reaches stability we can easily enough mask the internal server error internals.